Kernel/Devices/HID/PS2/KeyboardDevice.cpp

```diff
@@ -47,7 +47,23 @@ UNMAP_AFTER_INIT ErrorOr<NonnullOwnPtr<PS2KeyboardDevice>> PS2KeyboardDevice::tr
 
 UNMAP_AFTER_INIT ErrorOr<void> PS2KeyboardDevice::initialize()
 {
-    return attached_controller().reset_device(attached_port_index());
+    ErrorOr<void> err = attached_controller().reset_device(attached_port_index());
+
+    if (err.is_error()) {
+        TRY(attached_controller().send_command(attached_port_index(), SerialIOController::DeviceCommand::GetDeviceID));
+        ErrorOr<u8> res_err = attached_controller().read_from_device(attached_port_index());
+        if (res_err.is_error()) {
+            return err;
+        }
+        u8 res = res_err.value();
+        if (res == 0xab || res == 0xac || /* Regular and NCD Sun keyboards */
```

Comment:
It's probably better to use a switch-case statements here and then get rid of the C style comments as well (so you can use `//` comments).

Reply:
Done